tests/functional/SLM.php

```diff
@@ -786,4 +788,188 @@ public function testComputationByMonth()
+            "entities_id"  => $entity,
+            "calendars_id" => 0, // No specific calendar
+        ]);
+
+        // Create SLAs
+        $sla_tto = $this->createItem("SLA", [
+            "slms_id"         => $slm->getID(),
+            "name"            => "Test SLA tto",
+            "entities_id"     => $entity,
+            "type"            => \SLM::TTO,
+            "number_time"     => 4,
+            "definition_time" => "hour",
+        ]);
+        $sla_ttr = $this->createItem("SLA", [
+            "slms_id"         => $slm->getID(),
+            "name"            => "Test SLA ttr",
+            "entities_id"     => $entity,
+            "type"            => \SLM::TTR,
+            "number_time"     => 12,
+            "definition_time" => "hour",
+        ]);
+
+        // Create OLAs
+        $ola_tto = $this->createItem("OLA", [
+            "slms_id"         => $slm->getID(),
+            "name"            => "Test OLA tto",
+            "entities_id"     => $entity,
+            "type"            => \SLM::TTO,
+            "number_time"     => 2,
+            "definition_time" => "hour",
+        ]);
+        $ola_ttr = $this->createItem("OLA", [
+            "slms_id"         => $slm->getID(),
+            "name"            => "Test OLA ttr",
+            "entities_id"     => $entity,
+            "type"            => \SLM::TTR,
+            "number_time"     => 8,
+            "definition_time" => "hour",
+        ]);
+
+        // Create one escalation level for each SLA and OLA
+        $this->createItems("SlaLevel", [
+            [
+                "slas_id"        => $sla_tto->getID(),
+                "name"           => "Test escalation level (SLA/TTO)",
+                "entities_id"    => $entity,
+                "execution_time" => -900, // 15 minutes
+                "is_active"      => true,
+            ],
+            [
+                "slas_id"        => $sla_ttr->getID(),
+                "name"           => "Test escalation level (SLA/TTR)",
+                "entities_id"    => $entity,
+                "execution_time" => -1800, // 30 minutes
+                "is_active"      => true,
+            ],
+        ]);
+        $this->createItems("OlaLevel", [
+            [
+                "olas_id"        => $ola_tto->getID(),
+                "name"           => "Test escalation level (OLA/TTO)",
+                "entities_id"    => $entity,
+                "execution_time" => -2700, // 45 minutes
+                "is_active"      => true,
+            ],
+            [
+                "olas_id"        => $ola_ttr->getID(),
+                "name"           => "Test escalation level (OLA/TTR)",
+                "entities_id"    => $entity,
+                "execution_time" => -3600, // 60 minutes
+                "is_active"      => true,
+            ],
+        ]);
+
+        // Create a ticket 1 hour ago without any SLA
+        $date_1_hour_ago = date('Y-m-d H:i:s', strtotime('-1 hour'));
```

Comment:
The tests failed because you are calculating the `$date_1_hour_ago` relative to `now`. You should calculate it relative to `$_SESSION['glpi_currenttime']`.

Something like:
```suggestion
        $date_1_hour_ago = date('Y-m-d H:i:s', strtotime('-1 hour', strtotime($_SESSION['glpi_currenttime'])));
```